File: practice15/7.cpp
//https://www.nowcoder.com/practice/e2422543519249f292d8435394ab82fe?tpId=3&tqId=10872&tPage=1&rp=&ru=/ta/hackathon&qru=/ta/hackathon/question-ranking
#include<iostream>
#include<string>
using namespace std;
int main()
{
	string str;
	long long i, m, k, sum, l;
	while (cin >> str)
	{
		l = str.size();
		sum = 0;
		for (i = 0; i < l; i++)
		{
			sum = sum + str[i] - '0';
		}
		m = sum;
		sum = 0;
		while (m > 0)
		{
			k = m % 10;
			sum = sum + k;
			m = m / 10;
		}
		while (sum >= 10)
		{
			m = sum;
			sum = 0;
			while (m > 0)
			{
				k = m % 10;
				sum = sum + k;
				m = m / 10;
			}
		}
		cout << sum << endl;
	}
	system("pause");
	return 0;
}

File: practice31/7.java


//https://www.nowcoder.com/practice/c0ccc2e437f0473e8e5ebb7703c24089?tpId=90&tqId=30870&tPage=5&rp=5&ru=/ta/2018test&qru=/ta/2018test/question-ranking
import java.io.*;

public class Main {
    public static void main(String args[]) throws IOException {
        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(System.in));
        String S = bufferedReader.readLine();
        String T = bufferedReader.readLine();
        int x = S.length() - T.length();
        int sum = 0;
        StringBuilder str = new StringBuilder(T);
        StringBuilder temp;
        for (int i = 0; i <= x; i++) {
            temp = new StringBuilder(S.substring(i, i+T.length()));
            sum = sum + dist(temp, str);
        }
        System.out.println(sum);
    }

    public static int dist(StringBuilder str1, StringBuilder str2) {
        int count = 0;
        for (int i = 0; i < str1.length(); i++) {
            if (str1.charAt(i) != str2.charAt(i)) {
                count++;
            }
        }
        return count;
    }
}